src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -592,6 +592,19 @@ private bool PackHasChanged(WorkloadPack oldPack, WorkloadPack newPack)
             return false;
         }
 
+        /// <summary>
+        /// Finds the manifest for a specified workload.
+        /// </summary>
+        /// <param name="workloadId">The workload Id for which we want the corresponding manifest.</param>
+        /// <returns>The manifest for a corresponding workload.</returns>
+        /// <remarks>
+        /// Will fail if the workloadId provided is invalid.
+        /// </remarks>
+        public WorkloadManifest GetManifestFromWorkload(WorkloadId workloadId)
+        {
+            return _workloads[workloadId].manifest;
```

Comment:
I think I'm okay having this throw an exception. Perhaps just add the exceptions to the doc comments

```cs
<exception>KeyNotFoundException</exception>
<exception>ArgumentNullException</exception>
```

So it's clear this behaves like a dictionary. The other Option would be to expose the `_workloads` as an IReadOnlyDictionary property, that way the user knows its accessing an actual dictionary

Reply:
Adding those exceptions in, thanks for pointing that out. I like the idea of exposing `_workloads` to make it more clear. But since it is an implementation detail of `WorkloadResolver` and not `IWorkloadResolver` I think it would require changes to the interface which I'm not sure we want.